rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -311,6 +314,72 @@ void debugger_frame::keyPressEvent(QKeyEvent* event)
 	{
 		switch (event->key())
 		{
+		case Qt::Key_D:
+		{
+			if (cpu->id_type() == 2 && g_cfg.core.spu_debug && !cpu->is_stopped())
+			{
+				input_dialog dlg(4, "", tr("Max MFC cmds logged"),
+					tr("Decimal only, max allowed is 1820."), "0", this);
+
+				QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+				mono.setPointSize(8);
+				dlg.set_input_font(mono, false);
+				dlg.set_clear_button_enabled(false);
+				dlg.set_button_enabled(QDialogButtonBox::StandardButton::Ok, false);
+				dlg.set_validator(new QRegExpValidator(QRegExp("^[1-9][0-9]*$")));
+
+				connect(&dlg, &input_dialog::text_changed, [&](const QString& changed)
+				{
+					u32 dummy = 0;
+					const std::string text = changed.toStdString();
+					std::from_chars(&text[0], &text.back() + text.size(), dummy, 10);
```

Comment:
just wondering why you don't use toUInt. Seems like the obvious choice in this case.

```
bool ok;
const u32 dummy = changed.toUInt(&ok, 10);
dlg.set_button_enabled(QDialogButtonBox::StandardButton::Ok, ok && dummy && dummy<= (SPU_LS_SIZE / sizeof(mfc_cmd_dump)));
```

Reply:
Done.